api/src/main/java/jakarta/data/metamodel/NumericAttribute.java

```diff
@@ -21,10 +21,47 @@
 
 import java.util.Objects;
 
+/**
+ * <p>Represents a numeric entity attribute in the {@link StaticMetamodel}.
+ * Numeric entity attributes can be sorted on in query results and can be
+ * compared against values in query restrictions. They can also be used as
+ * and within numeric expressions involving various arithmetic operations.
+ * </p>
+ *
+ * <p>Entity attribute types that are considered numeric include:</p>
+ *
+ * <ul>
+ * <li>numeric primitve types: {@code byte}, {@code double}, {@code float},
+ *     {@code int}, {@code long}, {@code short}</li>
+ * <li>numeric wrapper types: {@link Byte}, {@link Double}, {@link Float},
+ *     {@link Integer}, {@link Long}, {@link Short}</li>
+ * <li>{@link java.math.BigDecimal} and {@link java.math.BigInteger}</li>
+ * </ul>
+ *
+ * <p>Where possible, {@code NumericAttribute}, which provides more function,
+ * is preferred over {@link ComparableAttribute} and {@link SortableAttribute}.
+ * </p>
+ *
+ * @param <T> entity class of the static metamodel.
+ * @param <N> type of entity attribute (or wrapper type if primitive).
+ */
+
 public interface NumericAttribute<T,N extends Number & Comparable<N>>
         extends BasicAttribute<T,N>, NumericExpression<T, N> {
```

Comment:
Actually, `NumericAttribute` is supposed to extends `ComparableAttribute`.

```suggestion
        extends ComparableAttribute<T,N>, NumericExpression<T, N> {
```

Reply:
I assumed that it already did that, but didn't look closely enough.